### PR TITLE
feat: build debug images with symbols enabled for releases and ad-hoc

### PR DESCRIPTION
**Pull request by rbarabas**
## Describe your changes
Adds the option to compile our binaries with debugging symbols. This is currently only enabled for the `server` binaries. 

Removes old, duplicate CI job that created ad-hoc *images* for PRs which were also called debug images but had no debug symbols in the binaries. Now ad-hoc image builds are created with debugging symbols by default.

Features
- delve added to the `build` image
- `WITH_DEBUG` enables compiling with debugging symbols

## How best to test these changes
Check output of the build action

## Issue ticket number and link
https://tigrisdata.atlassian.net/browse/TIG-1550

----
```
❯ WITH_DEBUG=1 make server
make -C api/proto generate GEN_DIR=../../api/server/v1 API_DIR=..
make[1]: Nothing to be done for `generate'.
...
CGO_ENABLED=1 go build -tags=release -ldflags "-X 'github.com/tigrisdata/tigris/util.Version=6b8ca30' -X 'github.com/tigrisdata/tigris/util.BuildHash=6b8ca30'"  -gcflags "all=-N -l" -o server/service ./server
```

### CODE REVIEW

**Review by JigarJoshi — DISMISSED** [review]

File: .github/workflows/push-docker-debug.yaml [hunk 1]
```diff
@@ -1,18 +1,86 @@
...
+      - name: Build and push debug images to DockerHub
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          file: docker/Dockerfile
+          platforms: linux/amd64,linux/arm64
+          push: ${{ github.event_name != 'pull_request' }}
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+          build-args: ${{ inputs.build_args }}
```
Comment (efirs) [inline .github/workflows/push-docker-debug.yaml:86]: You need to add this in `.github/workflows/push-docker-image.yaml`.

In this file we only call `.github/workflows/push-docker-image.yaml` and pass build_args like this:
```
jobs:
  debug:
    if: ${{ github.event.label.name == 'debug image' }}
    uses: ./.github/workflows/push-docker-image.yaml
    secrets: inherit
    with:
      docker_repository: "tigrisdata/tigris-debug"
      quay_repository: "quay.io/tigrisdata/tigris-debug"
      file: docker/Dockerfile
      platforms: amd64
      event_name: push
      build_args: WITH_DEBUG=1
```

Comment (rbarabas) [reply .github/workflows/push-docker-debug.yaml:86]: Let me take a look. I admit that I haven't reviewed the new structure in place to see what has been standardized and deduplicated. Stay tuned!

Comment (rbarabas) [reply .github/workflows/push-docker-debug.yaml:86]: OK so on publish we seem to run `push-docker-image` with arguments like platforms and you want me to add this not as standalone build but rather part of the base `push-docker-image` and call it with the extra argument?

Comment (rbarabas) [reply .github/workflows/push-docker-debug.yaml]: Please review again, modified based on above.

**Review by pboros — APPROVED** [review]

**Review by pboros — APPROVED** [review]

**Pull request merged**